zingo-memo/src/memo_serde.rs

```diff
@@ -0,0 +1,156 @@
+use std::io;
+
+use crate::{
+    utils::{
+        read_transaction_relative_height_and_index, read_unified_address_from_raw_encoding,
+        recover_absolute_transaction_heights, write_transaction_height_and_index,
+        write_unified_address_to_raw_encoding,
+    },
+    ParsedMemo,
+};
+use zcash_client_backend::address::UnifiedAddress;
+use zcash_encoding::{CompactSize, Vector};
+use zcash_primitives::consensus::BlockHeight;
+
+/// Packs a list of UAs into a memo. The UA only memo is version 0 of the protocol
+/// Note that a UA's raw representation is 1 byte for length, +21 for a T-receiver,
+/// +44 for a Sapling receiver, and +44 for an Orchard receiver. This totals a maximum
+/// of 110 bytes per UA, and attempting to write more than 510 bytes will cause an error.
```

Comment:
Why 510?  ` max(44X + 21Y+1)  <  511` where:  X,Y --> [0, 1, 2,...] ?

Reply:
512, -1 byte for arbitrary-memo, -1 byte for zingo-memo-version.